documentation-site/pages/theming/understanding-overrides.mdx

```diff
@@ -6,62 +6,222 @@ LICENSE file in the root directory of this source tree.
+      style: ({$theme}) => ({
+        color: $theme.colors.negative600,
+      }),
+    },
+  }}
+/>
+```
+
+## State Props
+
+The prop `$theme` is not the only variable that you can use in your style function. Most of subcomponents get various state props. For example, **the `Label` comes with**:
+
+- `$isDragged: boolean` - `true` if the list item is dragged
+- `$isSelected: boolean` - `true` if the list item is selected (space key-press)
+- `$isRemovable: boolean` - `true` if the list item is removable
+- `$value: React.Node` - item's value
+- `$index: number` - item's index
+
+Let's use `$isDragged` to change the Label color when dragged:
+
+```jsx
+<StatefulList
+  initialState={{
+    items: ['Item 1', 'Item 2', 'Item 3'],
+  }}
+  overrides={{
+    Label: {
+      style: ({$theme, $isDragged}) => ({
+        color: $isDragged ? $theme.colors.primary : $theme.colors.negative600,
+      }),
+    },
+  }}
+/>
+```
+
+And the result is that the Label gets blue when being dragged:
+
+<DndListExampleStateProps />
+
+## Overrides Inspector
+
+Almost every Base UI component has multiple overrides. **How can you learn what's available to you?** Every component page has the `Overrides` section at the bottom. It lists all overridable subcomponents and highlights each one of them. This box is taken straight from the [dnd-list page](/components/dnd-list/#overrides):
+
+<Overrides
+  name="Drag and Drop List"
+  component={ListExports}
+  renderExample={props => (
+    <StatefulList
+      removable
+      initialState={{items: ['Item 1', 'Item 2', 'Item 3', 'Item 4']}}
+      overrides={props.overrides}
+    />
+  )}
+/>
+
+If you are interested in **what state props your style functions can use**, you'll find the list of them at the bottom of every component page. Scroll down to the API section and click on the `Expand Prop Shape` button for the `overrides` property.
+
+## Override The Whole Subcomponent
+
+> **This is a very advanced technique and rarely needed**. If you go down this path, you might also need to inspect our source code to fully understand all behaviors that subcomponents should/can implement.
+
+So far we demonstrated how to **override styles or add additional props** but you can also **completely replace subcomponents**. This means you can alter the behavior and appearance of all Base UI components. For example, **we can enhance our textual Label and add a cloning functionality**:
+
+```jsx
+import React from 'react';
+import {List, arrayMove} from 'baseui/dnd-list';
+
+export default class Example extends React.Component {
+  state = {
+    items: ['Car', 'Truck', 'Bike', 'Skateboard'],
+  };
+  render() {
+    return (
+      <List
+        items={this.state.items}
+        onChange={({oldIndex, newIndex}) =>
+          this.setState(prevState => ({
+            items: arrayMove(prevState.items, oldIndex, newIndex),
+          }))
+        }
+        overrides={{
+          Label: {
+            component: ({$value}) => (
```

Comment:
We should move the creation of an override component out of the render for a best practices example

Reply:
Do you want to open a PR? 😄